aws/subscriber.go

```diff
@@ -79,7 +84,7 @@ func (s *awsSubscriber) Start(ctx context.Context, opts ...pubsub.Option) (<-cha
 			case <-ctx.Done():
 				return
 			default:
-				messages, err := s.pull(ctx)
+				messages, err := s.pull(ctx, decoder)
```

Comment:
Why decoder passed to pull function, you can just use option pattern which already defined for pubsub.Subscriber interface